plugins/woocommerce-blocks/assets/js/base/components/combobox/index.tsx

```diff
@@ -88,72 +145,175 @@ const Combobox = ( {
 		setValidationErrors,
 	] );
 
-	// @todo Remove patch for ComboboxControl once https://github.com/WordPress/gutenberg/pull/33928 is released
-	// Also see https://github.com/WordPress/gutenberg/pull/34090
+	const outerWrapperClasses = classnames(
+		'wc-block-components-combobox',
+		className || '',
+		{
+			'is-active': selectedOption?.value,
+			'has-error': error?.message && ! error?.hidden,
+		}
+	);
+
+	const ariaInvalid = error?.message && ! error?.hidden ? 'true' : 'false';
+	const store = useComboboxStore();
+	const activeValue = store.useState( 'activeValue' );
+
+	const onClose = useCallback( () => {
+		// If the search term doesn't match the selected option, try do an exact value match.
+		// e.g. if the user leaves "NZ" in the search box, select "New Zealand" on close.
+		// If not just leave the last selected value.
+		if ( searchTerm && selectedOption?.label !== searchTerm ) {
+			const exactValueMatch = findExactMatchBy(
+				'value',
+				searchTerm,
+				options
+			);
+
+			if ( exactValueMatch ) {
+				setSelectedOption( exactValueMatch );
+				onChange( exactValueMatch.value );
+				setSearchTerm( exactValueMatch.label );
+			} else {
+				setSearchTerm( selectedOption?.label || '' );
+			}
+		} else {
+			setSearchTerm( selectedOption?.label || '' );
+		}
+	}, [ setSearchTerm, selectedOption, searchTerm, onChange, options ] );
+
+	const minHeight =
+		matchingSuggestions.length * 35 > 300
+			? 300
+			: matchingSuggestions.length * 35;
+
 	return (
 		<div
-			id={ controlId }
-			className={ classnames( 'wc-block-components-combobox', className, {
-				'is-active': value,
-				'has-error': error?.message && ! error?.hidden,
-			} ) }
 			ref={ controlRef }
+			id={ controlId }
+			className={ outerWrapperClasses }
 		>
-			<ComboboxControl
-				className={ 'wc-block-components-combobox-control' }
-				label={ label }
-				onChange={ onChange }
-				onFilterValueChange={ ( filterValue: string ) => {
-					if ( filterValue.length ) {
-						// If we have a value and the combobox is not focussed, this could be from browser autofill.
-						const activeElement = isObject( controlRef.current )
-							? controlRef.current.ownerDocument.activeElement
-							: undefined;
-
-						if (
-							activeElement &&
-							isObject( controlRef.current ) &&
-							controlRef.current.contains( activeElement )
-						) {
-							return;
-						}
```

Comment:
Is there anything that's matching this behavior?

Reply:
@senadir No, and as far as I could tell without it it wasn't making autofill worse (although maybe you disagree from your testing, I'll need to look closer at that). My plan was to focus on the autofill bugs in a follow up PR.

Reply:
I've improved this now. I have specific handling for the autocomplete situation by checking if the input is in focus. In testing this worked for 1password autofill as well, but would love for you to take another look @senadir